### PR TITLE
feat(multi actions): add use-tools dust app

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

https://github.com/dust-tt/dust/issues/4602

This PR adds the new "use-tools" dust app (https://dust.tt/w/78bda07b39/a/0e9889c787), which is similar to generate-inputs except that it also picks what tool to use. Also added a small function that calls the dust app (had to write it to test the dust app)

Right now, it kind of expects a generation config (otherwise we won't have a "reply to user" action and it will be confusing for the model).

But all of this will get fine-tuned once we have a proper loop 👍 

## Risk

Some critical path stuff, but overall well tested (including the new unused dust app) and works well.

## Deploy Plan

N/A

### CODE REVIEW

**Review by spolu — APPROVED** [review]
LGTM

File: front/lib/api/assistant/agent.ts [hunk 1]
```diff
@@ -78,3 +103,181 @@ function isLegacyAgent(configuration: AgentConfigurationType): boolean {
...
+    "You are a conversational assistant with access to function calling."
+  );
+
+  // TODO(@fontanierh): revisit
+  const model = GPT_4_TURBO_MODEL_CONFIG;
+
+  const MIN_GENERATION_TOKENS = 2048;
+
+  // Turn the conversation into a digest that can be presented to the model.
+  const modelConversationRes = await renderConversationForModel({
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/agent.ts:136]: Yep for sure

File: types/src/front/lib/actions/registry.ts [hunk 2]
```diff
@@ -214,6 +214,22 @@ export const DustProdActionRegistry = createActionRegistry({
...
+    app: {
+      workspaceId: PRODUCTION_DUST_APPS_WORKSPACE_ID,
+      appId: "0e9889c787",
+      appHash:
+        "e2be0e5ee990a6ef86d1a89c8b55f924ff93dd6daaa0a502984d63b159c65ffd",
+    },
+    config: {
+      MODEL: {
+        provider_id: "openai",
+        model_id: GPT_4_TURBO_MODEL_ID,
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply types/src/front/lib/actions/registry.ts:227]: Agreed, changing to `GPT_4_TURBO_MODEL_CONFIG.modelId`

File: front/lib/api/assistant/actions/dust_app_run.ts [hunk 3]
```diff
@@ -93,8 +100,8 @@ export async function dustAppRunActionSpecification(
   }
 
   return new Ok({
-    name: appName,
-    description: appDescription || "",
+    name: name ?? appName,
+    description: description ?? appDescription ?? "",
```
Comment (flvndvd) [inline front/lib/api/assistant/actions/dust_app_run.ts:104]: Can we do this on the lines 68 and 69?

File: front/lib/api/assistant/agent.ts [hunk 4]
```diff
@@ -78,3 +103,181 @@ function isLegacyAgent(configuration: AgentConfigurationType): boolean {
...
+        `Error running use-tools action: [${res.error.type}] ${res.error.message}`
+      )
+    );
+  }
+
+  const { eventStream } = res.value;
+
+  const output: {
+    name?: string;
+    arguments?: Record<string, string | boolean | number>;
```
Comment (flvndvd) [inline front/lib/api/assistant/agent.ts:235]: `nit` create a proper type for this return type?

**Pull request merged**